File: Napakalaki.java/src/NapakalakiGame/Treasure.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author victor
 */
public class Treasure {
    private String name;
    private int bonus;
    private TreasureKind type;
    
    public Treasure(String n, int b, TreasureKind t){
        name = n;
        bonus = b;
        type = t;
    }

    public String getName(){
       return name; 
    }

    public int getBonus(){
        return bonus;
    }

    public TreasureKind getType(){
        return type;
    }
    
     @Override
    public String toString(){
        
        return name + "\n" + "+" + bonus + "\n" + type + "\n";
    }
}

File: Napakalaki.ruby/Treasure.rb
#encoding: utf-8
require_relative 'TreasureKind'

module NapakalakiGame

class Treasure

  def initialize(n, b, t)
    @name = n
    @bonus = b
    @type = t
  end

   def getName
     @name
   end

   def getBonus
     @bonus
   end

   def getType
     @type
   end

   def to_s
     "#{@name}" + "\n" + "+#{@bonus}" + "\n" + "#{@type}" + "\n"
   end
end

end
